packages/wdio-junit-reporter/src/index.js

```diff
@@ -160,7 +160,7 @@ class JunitReporter extends WDIOReporter {
 
         if (runner.config.hostname.indexOf('browserstack') && runner.capabilities.os === 'Android') {
           // NOTE: deviceUUID is used resulting in a everchanging package name in runner.sanitizedCapabilities when running Android tests under Browserstack. (i.e. ht79v1a03938.android.9)
-          let browserstackSanitizedCapabilities = runner.capabilities.device.replace(' ', '') + '.' + runner.capabilties.os + '.' + runner.capabilities.os_version.replace(' ', '_');
+          const browserstackSanitizedCapabilities = (runner.capabilities.device + '.' + runner.capabilties.os + '.' + runner.capabilities.os_version).replace(' ', '');
```

Comment:
Are you sure `runner.capabilities.device`, `runner.capabilties.os` and `runner.capabilities.os_version` are always present?